### PR TITLE
Add centos8-devel box for Katello devel environments on EL8

### PR DESCRIPTION
**Pull request by ehelms**
Requires https://github.com/theforeman/puppet-katello_devel/pull/245

This also requires https://github.com/theforeman/forklift/pull/1217 as the base CentOS 8 box is only 10GB and the devel install will run out of disk space before completing.

### CODE REVIEW

File: vagrant/boxes.d/99-local.yaml.example [hunk 1]
```diff
@@ -10,6 +10,20 @@ centos7-katello-devel:
...
+  primary: true
+  box: centos8
+  disk_size: 40
+  ansible:
+    playbook: 'playbooks/katello_devel.yml'
+    group: 'devel'
+    variables:
+      ssh_forward_agent: true
+      foreman_devel_github_push_ssh: True
+      katello_devel_github_username: ehelms
```
Comment (ekohl) [inline vagrant/boxes.d/99-local.yaml.example:23]: Can we use a more generic example?
```suggestion
      katello_devel_github_username: <REPLACE ME>
```

File: vagrant/.bundle/config [hunk 2]
```diff
@@ -0,0 +1,2 @@
+---
+BUNDLE_PATH: "/home/ehelms/.rubygems/"
```
Comment (ekohl) [inline vagrant/.bundle/config:2]: This looks unrelated

File: roles/katello_devel/meta/main.yml [hunk 3]
```diff
@@ -6,15 +6,19 @@ dependencies:
     foreman_server_repositories_katello: true
     foreman_server_repositories_foreman_client: true
   - role: ruby_scl
+    when: ansible_distribution_major_version == "7"
   - role: nodejs_scl
+    when: ansible_distribution_major_version == "7"
   - role: postgresql_scl
+    when: ansible_distribution_major_version == "7"
```
Comment (evgeni) [inline roles/katello_devel/meta/main.yml:13]: should this instead become a `devel_scls` meta-role? (not blocking this PR)

File: vagrant/boxes.d/99-local.yaml.example [hunk 4]
```diff
@@ -10,6 +10,20 @@ centos7-katello-devel:
       foreman_devel_github_push_ssh: True
       katello_devel_github_username: <REPLACE ME>
 
+centos8-devel:
```
Comment (evgeni) [inline vagrant/boxes.d/99-local.yaml.example:13]: technically `centos8-katello-devel` right, to match the other box names?

File: vagrant/boxes.d/99-local.yaml.example [hunk 5]
```diff
@@ -10,6 +10,20 @@ centos7-katello-devel:
...
+  primary: true
+  box: centos8
+  disk_size: 40
```
Comment (evgeni) [inline vagrant/boxes.d/99-local.yaml.example:16]: should this be instead set on the `centos8` box directly, as it will also be required for the pipeline boxes?

File: roles/katello_devel/meta/main.yml (hunk 3 above)
Comment (ehelms) [reply roles/katello_devel/meta/main.yml:13]: Maybe? I do not see the need for that additional layer of indirection presently.

**Review by evgeni — APPROVED** [review]

File: roles/foreman_installer_devel_scenario/tasks/main.yml [hunk 6]
```diff
@@ -15,6 +15,12 @@
     dest: /etc/foreman-installer/scenarios.d
   with_items:
     - katello-devel.yaml
```
Comment (ekohl) [inline roles/foreman_installer_devel_scenario/tasks/main.yml:17]: This is now just a single file. Mind changing the name to `Copy scenario file`?

File: roles/foreman_installer_devel_scenario/templates/katello-devel-answers.yaml [hunk 7]
```diff
@@ -6,11 +6,8 @@ certs:
...
   deployment_dir: /home/vagrant
   user: vagrant
-  db_type: postgres
-  rvm: true
 foreman_proxy_content:
-  pulp_master: true
-  qpid_router_broker_addr: localhost
+  proxy_pulp_yum_to_pulpcore: true
 foreman_proxy:
   custom_repo: true
```
Comment (ekohl) [inline roles/foreman_installer_devel_scenario/templates/katello-devel-answers.yaml:12]: If you want to clean up: this parameter is gone from the module. tftp is also off by default nowadays, same for templates.

**Review by ekohl — APPROVED** [review]

**Comment by ehelms** [comment]
Thanks for approvals. I will wait to merge this till https://github.com/theforeman/puppet-katello_devel/pull/245 is merged since it will give a false sense of availability until then.
